packages/app/src/cli/services/context.ts

```diff
@@ -167,37 +167,37 @@ export async function ensureDevContext(options: DevContextOptions): Promise<DevC
   developerPlatformClient = remoteApp?.developerPlatformClient ?? developerPlatformClient
 
   let orgId = getOrganization() || cachedInfo?.orgId
-  if (!orgId) {
+  if (!orgId || !developerPlatformClient) {
     const org = await selectOrg()
     developerPlatformClient = selectDeveloperPlatformClient({organization: org})
     orgId = org.id
   }
 
-  let {app: selectedApp, store: selectedStore} = await fetchDevDataFromOptions(options, orgId, developerPlatformClient)
   const organization = await fetchOrgFromId(orgId, developerPlatformClient)
 
+  // if we have selected an app or a dev store from a command flag, we keep them
+  let {app: selectedApp, store: selectedStore} = await fetchDevDataFromOptions(options, orgId, developerPlatformClient)
+
   if (!selectedApp || !selectedStore) {
-    // if we have selected an app or a dev store from a command flag, we keep them
     // if not, we try to load the app or the dev store from the current config or cache
     // if that's not available, we prompt the user to choose an existing one or create a new one
-    const [_selectedApp, _selectedStore] = await Promise.all([
-      selectedApp ||
-        remoteApp ||
+    const [cachedApp, cachedStore] = await Promise.all([
+      remoteApp ||
         (cachedInfo?.appId && appFromId({apiKey: cachedInfo.appId, organizationId: orgId, developerPlatformClient})),
```

Comment:
you need to keep `selectedApp` here, because this code path can be executed if `selectedStore` is not present, but `selectedApp` might!

Reply:
same for the `selectedStore`